pkg/provisioningv2/rke2/planner/etcdcreate.go

```diff
@@ -62,39 +54,23 @@ func (p *Planner) runEtcdSnapshotCreate(controlPlane *rkev1.RKEControlPlane, clu
 	return errs
 }
 
-func (p *Planner) generateEtcdSnapshotCreatePlan(controlPlane *rkev1.RKEControlPlane, snapshot *rkev1.ETCDSnapshotCreate, nodeName string) (plan.NodePlan, error) {
+func (p *Planner) generateEtcdSnapshotCreatePlan(controlPlane *rkev1.RKEControlPlane, entry *planEntry) (plan.NodePlan, error) {
 	args := []string{
 		"etcd-snapshot",
 	}
 
-	if snapshot.Name != "" {
-		args = append(args, fmt.Sprintf("--name=%s", snapshot.Name))
-	}
-	if nodeName != "" {
-		args = append(args, fmt.Sprintf("--node-name=%s", nodeName))
-	}
-
-	s3Args, s3Env, s3Files, err := p.etcdS3Args.ToArgs(snapshot.S3, controlPlane)
-	if err != nil {
-		return plan.NodePlan{}, err
-	}
-
 	return p.commonNodePlan(controlPlane, plan.NodePlan{
-		Files: s3Files,
-		Instructions: []plan.Instruction{{
-			Name:    "create",
-			Command: rke2.GetRuntimeCommand(controlPlane.Spec.KubernetesVersion),
-			Env:     s3Env,
-			Args:    append(args, s3Args...),
-		}},
+		Instructions: []plan.OneTimeInstruction{
+			p.generateInstallInstructionWithSkipStart(controlPlane, entry),
+			{
+				Name:    "create",
+				Command: rke2.GetRuntimeCommand(controlPlane.Spec.KubernetesVersion),
+				Args:    args,
+			}},
 	})
 }
 
 func (p *Planner) createEtcdSnapshot(controlPlane *rkev1.RKEControlPlane, clusterPlan *plan.Plan) []error {
-	if !rke2.Provisioned.IsTrue(controlPlane) && controlPlane.Status.ETCDSnapshotCreatePhase == "" {
```

Comment:
Yes. If a user attempts to take a snapshot of a cluster that is not in a full blown ready state, the snapshot create phase will go to "failed" and fallthrough to finish provisioning the rest of the cluster.

I removed that check because it blocks creating etcd snapshots if the cluster is in a broken state but still relatively functional, which is not ideal.